### DISCUSSION TITLE
audioplayererror [requesterror]: socket hang up got

### DISCUSSION THREAD

**Discussion by artemian4ik** [sindresorhus/got#2241] [Q&A] [answered]
#### Hi. I get problem with streaming music radio.
Get error: audioplayererror [requesterror]: socket hang up got

```
function startMusicRadio() {
...
code
...

const player = createAudioPlayer();
const stream = got.stream("https://radio.advance-rp.ru/channel1.ogg");
 const resource = createAudioResource(stream, {
        inputType: StreamType.OggOpus,
        inlineVolume: true,
});
resource.volume.setVolume(0.5);
player.play(resource);
const subscription = connection.subscribe(player);
player.on('error', error => {
        console.error(error);
        startMusicRadio();
});
```

How I can player start again if player get error?

#### Checklist

- [+] I have read the documentation.

**Comment by sindresorhus** [comment, marked as answer]
"Socket hang up" errors happen when the radio stream connection drops unexpectedly. The issue is that you need to handle the error on the Got stream itself, not just the audio player.
